src/notification/notificationmodel.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <QColor>
+#include <QAbstractListModel>
+
+/**
+ * @brief Notification model
+ *
+ */
+class NotificationModel : public QAbstractListModel
+{
+    Q_OBJECT
+public:
+    /**
+     * @brief Construct a new NotificationModel
+     *
+     * @param parent
+     */
+    NotificationModel(QObject* parent = nullptr);
+
+    enum {
+        Text,
+        Color,
+        Time,
+        Icon,
+        Severity,
+    };
+
+    /**
+     * @brief Return data
+     *
+     * @param index
+     * @param role
+     * @return QVariant
+     */
+    QVariant data(const QModelIndex& index, int role) const override;
+
+    /**
+     * @brief Set the data
+     *
+     * @param index Model
+     * @param value Value to set
+     * @param role Model role
+     * @return true Was able to set value
+     * @return false Not able to set value
+     */
+    bool setData(const QModelIndex& index, const QVariant& value, int role) override;
+
+    /**
+     * @brief Get role names
+     *
+     * @return QHash<int, QByteArray>
+     */
+    QHash<int, QByteArray> roleNames() const override;
+
+    /**
+     * @brief Return the number of rows
+     *
+     * @param parent
+     * @return int
+     */
+    int rowCount(const QModelIndex &parent = QModelIndex()) const override
+    {
+        Q_UNUSED(parent)
+        return _size;
+    }
+
+    /**
+     * @brief Inserts count rows into the model, beginning at the given row.
+     *  This implementation was based over QStringListModel
+     * @param row Row to be used
+     * @param count Number of times
+     * @param parent
+     * @return true If everything goes fine
+     * @return false It was not able to insert a new row
+     */
+    bool insertRows(int row, int count, const QModelIndex &parent = QModelIndex()) override;
+
+    /**
+     * @brief Add a new notification in the model
+     *
+     * @param text
+     * @param icon
+     * @param color
+     * @param severity
+     */
+    Q_INVOKABLE void add(
+        QVariant text = QStringLiteral("This is a empty notification."),
+        QVariant icon = {},
+        QVariant color = QColor("DarkGray"),
+        QVariant severity = {}
+    );
+
+    /**
+     * @brief Remove index in model
+     *
+     * @param index
+     */
+    Q_INVOKABLE void remove(int index);
+
+    /**
+     * @brief Remove all items
+     *
+     */
+    Q_INVOKABLE void removeAll();
+
+private:
+    Q_DISABLE_COPY(NotificationModel)
```

Comment:
Why is this necessary?

Reply:
http://doc.qt.io/qt-5/qobject.html#Q_DISABLE_COPY

I see now, but why don't we do this on NotificationManager or our other QObject derived classes?

Reply:
NotificationManager is a singleton, it's not possible to copy by it definition.
But about the other QObjects that we have yes. This line is a result based on Qt source code study of models that already exist.

Reply:
The document that I linked says that any QObject should never be copied, not just singletons.

Reply:
Check this line:
https://github.com/bluerobotics/ping-viewer/pull/315/files#diff-ef6adb74577178213a0ad3d437b8b25cR45

Reply:
hehe, it looks like this is the same effect and purpose of Q_DISABLE_COPY, so why not use that?

Reply:
Good point, I thought that was a different kind of disable copy but they are not. Will change it to make it uniform.